packages/lesswrong/lib/collections/users/schema.ts

```diff
@@ -498,11 +498,15 @@ const schema: SchemaType<DbUser> = {
   
   theme: {
     type: String,
-    optional: true, 
+    optional: true,
+    nullable: true,
     canCreate: ['members'],
     canUpdate: ownsOrIsAdmin,
     canRead: ownsOrIsAdmin,
-    hidden: true,
+    hidden: forumTypeSetting.get() !== "EAForum",
+    control: "ThemeSelect",
```

Comment:
Yeah, that's correct. Originally I added it so that we could preview the theme when the user selects a different option, but I think that's pretty confusing if it doesn't autosave (ie; user selects a different theme, the page updates to show the new theme, user navigates to the home page without submitting the form and the theme reverts to how it was before). I just left the component for now because maybe we want to do something in this space in the future.

WRT the general argument over the UI, I don't really love either of the two options:
 - I 100% agree that the user dropdown menu is too cluttered (plus we've had people saying recently that they think the fact that the username has a dropdown is pretty unintuitive - I can understand Habryka's reasoning here about deterring brand new users from creating new posts, but I think it's perfectly reasonable that brand new users would want to change the theme).
 - Having it in the user settings it problematic because logged-out users can't access it, plus the janky-ness with needing to submit the form/not being able to preview.

Maybe something like this would be better (but prettier, because I just hacked this together in the console...)? It gets it out of the user menu to somewhere a new user might be more likely to look, and it would also be easier to implement autosave and preview.
![Screenshot 2022-10-25 at 13 19 41](https://user-images.githubusercontent.com/5075628/197771398-e1be9354-dcf0-4ad7-b571-333ded336235.png)

Reply:
Let's discuss in 1-1